### ISSUE TITLE
OperationFailure: must have $meta projection for all $meta sort keys 

### ISSUE THREAD

**Issue by quentinpraz** [pyeve/eve#1176]
### Expected Behavior

I'am using the _Full Text Search_ capability of MongoDB. To do so, my queries look like that
``
http://eve/example?where={"$text":{"$search":"test"}}&projection={"score":{"$meta":"textScore"}}&sort=[("score", {"$meta": "textScore"})]
``

This is working fine with Eve 0.7.10
Note that I set the `allow_unknown` property to `True`.  (#841)

### Actual Behavior
After the upgrade to Eve 0.8, I have the following error:
``` pytb
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python2.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python2.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python2.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python2.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python2.7/site-packages/eve/endpoints.py", line 56, in collections_endpoint
    response = get(resource, lookup)
  File "/usr/lib/python2.7/site-packages/eve/methods/common.py", line 297, in rate_limited
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/eve/auth.py", line 78, in decorated
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/eve/methods/common.py", line 1181, in decorated
    r = f(resource, **combined_args)
  File "/usr/lib/python2.7/site-packages/eve/methods/get.py", line 39, in get
    return get_internal(resource, **lookup)
  File "/usr/lib/python2.7/site-packages/eve/methods/get.py", line 112, in get_internal
    return _perform_find(resource, lookup)
  File "/usr/lib/python2.7/site-packages/eve/methods/get.py", line 206, in _perform_find
    for document in cursor:
  File "/usr/lib/python2.7/site-packages/pymongo/cursor.py", line 1189, in next
    if len(self.__data) or self._refresh():
  File "/usr/lib/python2.7/site-packages/pymongo/cursor.py", line 1104, in _refresh
    self.__send_message(q)
  File "/usr/lib/python2.7/site-packages/pymongo/cursor.py", line 982, in __send_message
    helpers._check_command_response(first)
  File "/usr/lib/python2.7/site-packages/pymongo/helpers.py", line 155, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
OperationFailure: must have $meta projection for all $meta sort keys
```

### Environment

* Python version: 2.7.14
* Eve version: 0.8
* PyMongo: 3.7.1

**Comment by nicolaiarocci** [comment]
By looking at your stack trace, the exception is raised by PyMongo. Did you try to backport it a little bit?

**Comment by quentinpraz** [comment]
I tested the _full text search_ query in two environments:
#### Environment 1

* Python version: 2.7.14
* Eve version: 0.8
* PyMongo: 3.7.1

&rarr; Working fine ✔️ 

#### Environment 2
* Python version: 2.7.14
* Eve version: 0.7.10
* PyMongo: 3.7.1

&rarr; Not working ❌ 

As you can see, the PyMongo version is the same.

**Comment by nicolaiarocci** [comment]
(in the previous post, the contrary appeared to be true)

**Comment by quentinpraz** [comment]
My bad, my second post was wrong (I updated it). It is not working with Eve 0.8. I had to roll back to version 7.10 to be able to use the _full text search_ query.

**Comment by nicolaiarocci** [comment]
Should be fixed in 0.8.1-dev
